### PR TITLE
Add org.gnome.FileRoller

### PR DESCRIPTION
**Pull request by A6GibKm**

### CODE REVIEW

**Comment by A6GibKm** [comment]
Upstream issue: https://gitlab.gnome.org/GNOME/file-roller/-/issues/87

**Comment by A6GibKm** [comment]
bot, build org.gnome.FileRoller

**Comment by A6GibKm** [comment]
bot, build org.gnome.FileRoller

**Comment by A6GibKm** [comment]
Created a PR upstream for the appdata: https://gitlab.gnome.org/GNOME/file-roller/-/merge_requests/43

**Comment by nedrichards** [comment]
blocking: there isn't an app icon in the about dialog. 
non-blocking:  the file preview/open functionality isn't the best integrated into the flatpak system

@bilelmoussaoui FYI with the above fixed I'm good to merge.

**Comment by A6GibKm** [comment]
> blocking: there isn't an app icon in the about dialog.

MIght be related to the icon being named as ArchiveManager originally?

**Comment by barthalion** [comment]
Bundling `unrar` and `p7zip` will make the app much more useful.

**Comment by A6GibKm** [comment]
Yes, I will do `7zip` now, but `unrar` might be less easy as it requires `extra-data` due to license issues, talking about that, it might be worth noticing that there are two apps on flathub which distribute it.

 I am not sure how to build it with, say `unrar`, I tried installing a `unrar` as a module but I was not able to find a difference.

**Comment by barthalion** [comment]
> `unrar` might be less easy as it requires `extra-data` due to license issues

Unrar license allows redistribution.

**Comment by A6GibKm** [comment]
Added unrar and p7zip, credits to @p1u3o [p1u3o/org.gnome.FileRoller](https://github.com/p1u3o/org.gnome.FileRoller).

Ok, I don't know the details but for all I know the unrar license is messy, I will leave it as extra-data for now.

**Comment by A6GibKm** [comment]
bot, build org.gnome.FileRoller

**Comment by barthalion** [comment]
Extra-data just leads to bad user experience and binds the application to external server instead of Flathub CDN. Unrar license is very clear that it can be distributed as part of other applications:

> The UnRAR sources may be used in any software to handle RAR
archives without limitations free of charge, but cannot be used
to re-create the RAR compression algorithm, which is proprietary.
Distribution of modified unRAR sources in separate form or as a
part of other software is permitted, provided that it is clearly
stated in the documentation and source comments that the code may
not be used to develop a RAR (WinRAR) compatible archiver.

**Comment by A6GibKm** [comment]
> Extra-data just leads to bad user experience and binds the application to external server instead of Flathub CDN. 

Agree.

As it is `7z` and `unrar` are in the path but `7za` and `7rz` are not being built. Is this Ok?

**Comment by A6GibKm** [comment]
bot, build org.gnome.FileRoller

**Comment by A6GibKm** [comment]
@nedrichards using `copy-icon` the issue you mention is fixed.

File: org.gnome.FileRoller.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
+{
+    "app-id" : "org.gnome.FileRoller",
+    "runtime" : "org.gnome.Platform",
+    "runtime-version" : "3.36",
+    "sdk" : "org.gnome.Sdk",
+    "command" : "file-roller",
+    "rename-icon": "org.gnome.ArchiveManager",
+    "copy-icon": true,
```
Comment (bilelmoussaoui) [inline org.gnome.FileRoller.json:8]: Hmm would be nice to rename those upstream. Feel free to open an issue regarding that.

Comment (A6GibKm) [reply org.gnome.FileRoller.json:8]: Will do, but I have the impression that it is more involved than that... the app uses the names FileRoller and ArchiveManager interchangeably.

Comment (A6GibKm) [reply org.gnome.FileRoller.json:8]: https://gitlab.gnome.org/GNOME/file-roller/-/issues/98

**Comment by p1u3o** [comment]
> > `unrar` might be less easy as it requires `extra-data` due to license issues
> 
> Unrar license allows redistribution.

The reason I used extra-data in my original manifest was that whilst `unrar` is open source (under restrictive licensing), `rar` itself is not.

IMHO the best path forward is to add an extension point for archive specific tools, perhaps a generic one that other applications can use, that's added to the flatpaks path. File Roller will pick these up.

**Pull request closed without merging**